core/src/main/scala/akka/kafka/scaladsl/Transactional.scala

```diff
@@ -86,6 +86,8 @@ object Transactional {
       transactionalId: String
   ): Flow[Envelope[K, V, ConsumerMessage.PartitionOffset], Results[K, V, ConsumerMessage.PartitionOffset], NotUsed] = {
     require(transactionalId != null && transactionalId.length > 0, "You must define a Transactional id.")
+    require(settings.producerFactory.isDefined, "You cannot use a shared or external producer factory.")
+    require(settings.producerFactoryAsync.isDefined, "You cannot use a shared or external async producer factory.")
```

Comment:
So this won't work with Akka Discovery? Isn't it OK as long as it creates a new instance? Guarding against `withProducer` is the important bit, right?

Reply:
This should work with Akka Discovery.  The internal `producerFactorySync` and `producerFactoryAsync` values are optional and only set when the user uses a `withProducer` method.  They're not used internally to create a producer by default.  We'll know for sure once the tests aren't blocked by other issues.

Reply:
Ah, you're right. I mixed up the `enrichAsync` and `producerFactoryAsync`.
A feature we tried to unlock with the producer factory was the possibility to wrap the producer with the [OpenTracing](https://github.com/opentracing-contrib/java-kafka-client) decorator.